### PR TITLE
Makes anomalies more dangerous

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
tweak: Anomalies move more often, are resistant to explosions and will only be destroyed if they are in devastation range. You shouldn't bomb them, though.
tweak: Hyper-energetic flux anomaly will shock mobs that run into it, or if it runs into them.
tweak: Bluespace anomaly will occasionally teleport mobs away from it in a small radius.
tweak: Vortex anomalies will sometimes throw objects at nearby living mobs.
tweak: Pyroclastic anomalies will produce bigger, hotter fires, and if not disabled, it will release an additional burst of flame. In addition, the resulting slime will be rabid and thus attack much more aggressively.
fix: Gravitational anomalies will now properly throw objects at nearby living things.
/:cl:

### CODE REVIEW

File: code/game/objects/effects/anomalies.dm [hunk 1]
```diff
@@ -22,10 +22,17 @@
 
 
 /obj/effect/anomaly/proc/anomalyEffect()
-	if(prob(50))
+	if(prob(70))
 		step(src,pick(alldirs))
 
 
+/obj/effect/anomaly/ex_act(severity, target)
+	switch(severity)
```
Comment (RemieRichards) [inline code/game/objects/effects/anomalies.dm]: a switch for one case is unnecessary

File: code/game/objects/effects/anomalies.dm [hunk 2]
```diff
@@ -22,10 +22,17 @@
...
-	if(prob(50))
+	if(prob(70))
 		step(src,pick(alldirs))
 
 
+/obj/effect/anomaly/ex_act(severity, target)
+	switch(severity)
+		if(1)
+			qdel(src)
+		else
```
Comment (RemieRichards) [inline code/game/objects/effects/anomalies.dm]: else, return is unnecessary

File: code/game/objects/effects/anomalies.dm [hunk 3]
```diff
@@ -22,10 +22,17 @@
...
+			return
```
Comment (phil235) [inline code/game/objects/effects/anomalies.dm]: the "else return" isn't really needed.

File: code/game/objects/effects/anomalies.dm [hunk 4]
```diff
@@ -22,10 +22,17 @@
 
 
 /obj/effect/anomaly/proc/anomalyEffect()
-	if(prob(50))
+	if(prob(70))
```
Comment (RemieRichards) [inline code/game/objects/effects/anomalies.dm]: make this a var, ```movechance```

File: code/game/objects/effects/anomalies.dm [hunk 5]
```diff
@@ -53,20 +60,18 @@
...
 	..()
-
 	boing = 1
 	for(var/obj/O in orange(4, src))
 		if(!O.anchored)
 			step_towards(O,src)
 	for(var/mob/living/M in orange(4, src))
 		step_towards(M,src)
-	for(var/obj/O in orange(1,src))
+	for(var/obj/O in range(0,src))
```
Comment (RemieRichards) [inline code/game/objects/effects/anomalies.dm]: ```orange()``` was a lazy way to make it not throw itself around.
if you don't want this, make sure to check for ```O``` being ```src```

File: code/modules/events/anomaly_flux.dm [hunk 6]
```diff
@@ -22,5 +22,5 @@
 
 /datum/round_event/anomaly/anomaly_flux/end()
 	if(newAnomaly.loc)//If it hasn't been neutralized, it's time to blow up.
-		explosion(newAnomaly, -1, 3, 8, 10)
+		explosion(newAnomaly, 1, 4, 16, 18) //Low devistation, but hits a lot of stuff.
```
Comment (phil235) [inline code/modules/events/anomaly_flux.dm]: "devastation"

File: code/game/objects/effects/anomalies.dm [hunk 7]
```diff
@@ -88,11 +93,39 @@
...
+	canshock = 1
+
+/obj/effect/anomaly/flux/Bump(mob/living/M)
+	mobShock(M)
+
+/obj/effect/anomaly/flux/Bumped(mob/living/M)
+	mobShock(M)
+
+/obj/effect/anomaly/flux/proc/mobShock(mob/living/M)
+	if(canshock && isliving(M))
```
Comment (phil235) [inline code/game/objects/effects/anomalies.dm]: Technically you could do `istype(M)`.

File: code/game/objects/effects/anomalies.dm (hunk 5 above)
Comment (Anonus) [inline code/game/objects/effects/anomalies.dm]: It's anchored, so it won't get itself as a target anyway.

**Comment by Metacide** [comment]
Sounds neat, most people don't really notice / care about anomalies at the moment. :+1:

**Comment by Supermichael777** [comment]
we need a dangerzone tag for when hazards are buffed

**Comment by Lumipharon** [comment]
Grav anomaly is already like the most lethal thing to naturally appear on the station.
It can chain stun and kill you in like 5 seconds.

**Comment by PKPenguin321** [comment]
you should be able to rig anomaly cores to a signaler (which you can already do) and have it so once they're signaled the anomaly pops out and becomes active again. that way we can make weaponized anomalies

**Comment by Lumipharon** [comment]
Neat idea

**Comment by Anonus** [comment]
That's a pretty cool idea, but it'd require an extensive rewrite of how anomalies work, spawn, and despawn and I'm not up to that right now!

Also if the grav anomaly turns out to be too deadly I'll probably tweak it a bit.

**Comment by Cheridan** [comment]
good changes

File: code/modules/events/anomaly_pyro.dm [hunk 8]
```diff
@@ -28,9 +28,12 @@
 
 /datum/round_event/anomaly/anomaly_pyro/end()
 	if(newAnomaly.loc)
-		explosion(get_turf(newAnomaly), -1,0,3, flame_range = 4)
+		var/turf/simulated/T = get_turf(newAnomaly)
+		if(istype(T))
+		T.atmos_spawn_air(SPAWN_HEAT | SPAWN_TOXINS | SPAWN_OXYGEN, 200) //Make it hot and burny for the new slime
```
Comment (tkdrg) [inline code/modules/events/anomaly_pyro.dm]: Botched indentation? This is throwing an empty if() warning. @Anonus

Comment (Anonus) [inline code/modules/events/anomaly_pyro.dm]: Do nearly everything right BUT NOT EVERYTHING ENOUGH

**Pull request merged**